reporter/src/main/kotlin/reporters/WebAppReporter.kt

```diff
@@ -45,10 +50,17 @@ class WebAppReporter : Reporter {
 
         val outputFile = outputDir.resolve(reportFilename)
 
-        outputFile.bufferedWriter().use {
-            it.write(prefix)
-            evaluatedModel.toJson(it, prettyPrint = false)
-            it.write(suffix)
+        outputFile.writeText(prefix)
+
+        FileOutputStream(outputFile, true).use {
+            val b64OutputStream = Base64OutputStream(it, true, 0, ByteArray(0))
+            val gzipWriter = GzipCompressorOutputStream(b64OutputStream).bufferedWriter()
+            evaluatedModel.toJson(gzipWriter, prettyPrint = false)
+            gzipWriter.close()
+        }
+
+        FileWriter(outputFile, true).use { writer ->
+            writer.write(suffix)
```

Comment:
The above together opens / closes the same file three times to write to it. I'd prefer if there was a way to open / close the file only once.

Reply:
I agree that it's not nice, but I couldn't get it to work correctly, the compressed data always ended up in the wrong place. I think that's something that can be fixed later, if possible.

Reply:
Maybe it's worth trying again with the above mentioned `wrap` function?

Reply:
Done. Didn't test with the additional Base 64 stream before, that probably fixed the issue under the hood.